### DISCUSSION TITLE
How can I generate types that are being used?

### DISCUSSION THREAD

**Discussion by baeharam** [dotansimha/graphql-code-generator#5644] [Q&A] [answered]
I have `schema.graphql` and generating types by graphql-code-generator. (I migrated from apollo-codegen) However, this tool is generating all the types in `schema.graphql`. Apollo-codegen generated types that were being used. Then, how can I generate types that are used only?

**Comment by dotansimha** [comment, marked as answer]
You can use `onlyOperationTypes: true` to get more clear results with `typescript-operations`. Combing it with `preResolveTypes: true` should give you a very clean output.
